cli/src/pcluster/config/cluster_config.py

```diff
@@ -2167,6 +2167,133 @@ def max_network_interface_count(self) -> int:
         return least_max_nics
 
 
+class InstanceRequirementsDefinition(Resource):
+    """Collects the attributes required to define a Slurm Compute Resource through Instance Requirements."""
+
+    def __init__(
+        self,
+        min_vcpus,
+        min_memory_mib,
+        max_vcpus=0,
+        max_memory_mib=0,
+        accelerator_count=0,
+        max_price_percentage=50,
+        allowed_instance_types: List[str] = None,
+        excluded_instance_types: List[str] = None,
+        accelerator_types: List[str] = None,
+        accelerator_manufacturers: List[str] = None,
+        bare_metal: List[str] = None,
+        instance_generations: List[str] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.min_vcpus = Resource.init_param(min_vcpus)
+        self.max_vcpus = Resource.init_param(max_vcpus)
+        self.min_memory_mib = Resource.init_param(min_memory_mib)
+        self.max_memory_mib = Resource.init_param(max_memory_mib)
+        self.accelerator_count = Resource.init_param(accelerator_count)
+        self.max_price_percentage = Resource.init_param(max_price_percentage)
+        self.allowed_instance_types = Resource.init_param(allowed_instance_types)
+        self.excluded_instance_types = Resource.init_param(excluded_instance_types)
+        self.bare_metal = bare_metal
+        self.instance_generations = instance_generations
+        self.accelerator_types = accelerator_types
+        self.accelerator_manufacturers = accelerator_manufacturers
+        # Enforce desired default behavior
+        if self.accelerator_count > 0:
+            self.accelerator_types = ["gpu"]
+            self.accelerator_manufacturers = ["nvidia"]
+        else:
```

Comment:
if you're forcing a specific behaviour you can avoid the set in the previous lines, that is useless:
```python
        self.accelerator_types = accelerator_types
        self.accelerator_manufacturers = accelerator_manufacturers
```